inc/classes/class-google-auth.php

```diff
@@ -90,26 +90,42 @@ protected function _get_client() {
 		$client->setClientId( $client_id );
 		$client->setClientSecret( $client_secret );
 
+		$client->setState( $this->_get_state() );
+
+		$login_url = $this->_get_login_url();
+
+		$client->setRedirectUri( $login_url );
+
+		return $client;
+
+	}
+
+	/**
+	 * Get the state to pass in OAuth.
+	 *
+	 * This state will be re-verified while authenticating the user.
+	 * Any modifications or tampering would result in no authentication.
+	 *
+	 * @return string
+	 */
+	protected function _get_state(): string {
 		$redirect_to = filter_input( INPUT_GET, 'redirect_to', FILTER_SANITIZE_URL );
 		$redirect_to = ( ! empty( $redirect_to ) ) ? $redirect_to : admin_url();
 
 		// If redirect_to url don't have host name then add that.
 		$redirect_to = ( ! wp_parse_url( $redirect_to, PHP_URL_HOST ) ) ? home_url( $redirect_to ) : $redirect_to;
 
-		$state = [
-			'redirect_to' => $redirect_to,
-			'blog_id'     => get_current_blog_id(),
-		];
-		$state = urlencode_deep( implode( '|', $state ) );
-
-		$client->setState( $state );
-
-		$login_url = $this->_get_login_url();
+		$state = apply_filters(
+			'wp_google_client_state',
```

Comment:
Recommend using `login_with_google/client_state` to be consistent with plugin name change since release. Otherwise, looks good.